asset/asset.go

```diff
@@ -1202,6 +1197,68 @@ func NewGKRCustomSubtreeRootRecord(root *chainhash.Hash) tlv.Record {
 	return tlv.MakePrimitiveRecord(GKRCustomSubtreeRoot, (*[32]byte)(root))
 }
 
+// NumsXPub turns the given NUMS key into an extended public key (using the x
+// coordinate of the public key as the chain code), then derives the actual key
+// to use from the derivation path 0/0. The extended key always has the mainnet
+// version, but can be converted to any network on demand by the caller with
+// CloneWithVersion().
+func NumsXPub(numsKey *btcec.PublicKey) (*hdkeychain.ExtendedKey,
```

Comment:
nit: Accept arg as just `numsKey btcec.PublicKey`. Why risk `nil` here?

I think this is just one of those patterns, but I don't see the point of it.